roles/host-provision/tasks/proxy.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+- name: proxy | Set fact
+  set_fact:
+    proxy_ip: "{{ ansible_env['SSH_CLIENT'].split() | first }}"
+  # debug:
+  #   var: proxy_ip
+  tags: host-provision
+
+- name: proxy | Validate proxy connectivity
+  uri:
+    url: https://www.google.com
+    status_code: 200
+  environment:
+    https_proxy: "http://{{ proxy_ip }}:3128"
+  tags: host-provision
+
+- name: proxy | Add gateway instance to /etc/hosts
+  become: true
+  lineinfile:
+    dest: /etc/hosts
+    regexp: '.*gateway-instance.*'
+    line: "{{ proxy_ip }} gateway-instance gateway-instance."
+    backup: yes    
+  tags: host-provision
+
+- name: proxy | Add proxy to /etc/profile.d
+  become: true
+  copy:
+    dest: /etc/profile.d/proxy.sh
+    content: |
+            export http_proxy="http://gateway-instance:3128"
+            export https_proxy="http://gateway-instance:3128"
+            export ftp_proxy="http://gateway-instance:3128"
+    backup: yes
+  tags: host-provision
+
+- name: proxy | Update proxy in sudoers
+  become: true
+  lineinfile:
+    dest: /etc/sudoers
+    regexp: ".*env_keep.*proxy.*"
+    line: 'Defaults env_keep += "ftp_proxy http_proxy https_proxy no_proxy"'
+    validate: 'visudo -cf %s'
+    backup: yes
+  tags: host-provision
+
+- name: proxy | Add repo
```

Comment:
EPEL is a bit tricky, as it's version dependent.  What about RHEL8, for example?  Oracle LInux?  Even SuSE?  Any background on what packages we need from EPEL, and if it's the best source (as opposed to, say, extras in RHEL https://access.redhat.com/solutions/912213)

Reply:
https://paste.googleplex.com/6545261326237696 (internal) shows the only iftop needs epel.

And based on: https://docs.google.com/document/d/1mX59YHO37MavCGmfLlTtEmitRc53oeYkEgU3PtYMbrg/edit (internal), EPEL is not needed. Taking it out.